### PR TITLE
Fix redirects on user settings introduced by menu refactoring

### PR DESCRIPTION
**Pull request by daviian**
Adresses https://github.com/go-gitea/gitea/issues/3974

Settings pages that host more than one modules now load missing data after `POST` requests.

### CODE REVIEW

**Comment by lafriks** [comment]
Doesn't changes won't be lost on redirect?

**Comment by daviian** [comment]
What changes do you mean? The flash messages?

**Comment by daviian** [comment]
@lafriks I had a look on how to implement that for all settings. However it is not a small change. So maybe we should postpone the prefilling of forms until this and the refactoring PR's are merged.

**Comment by daviian** [comment]
@lafriks @JonasFranzDEV I've changed the PR to just load missing data, but IMO this can be done better. But think this should be done in a separate PR and after all the other PR's regarding the settings are merged.

File: routers/user/setting/profile.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ const (
 func Profile(ctx *context.Context) {
 	ctx.Data["Title"] = ctx.Tr("settings")
 	ctx.Data["PageIsSettingsProfile"] = true
+
```
Comment (jonasfranz) [inline routers/user/setting/profile.go:35]: Why this change?

**Review by jonasfranz — CHANGES_REQUESTED** [review]

File: routers/user/setting/profile.go (hunk 1 above)
Comment (daviian) [reply routers/user/setting/profile.go:35]: Thought it is more consistent with the rest of the code base.
But of course I can remove it :)

**Review by techknowlogick — APPROVED** [review]

File: routers/user/setting/profile.go (hunk 1 above)
Comment (daviian) [reply routers/user/setting/profile.go:35]: @JonasFranzDEV Do you want me to remove it?

**Review by jonasfranz — APPROVED** [review]
